zebra-state/src/service.rs

```diff
@@ -52,7 +158,19 @@ impl Service<Request> for StateService {
 
     fn call(&mut self, req: Request) -> Self::Future {
         match req {
-            Request::CommitBlock { .. } => unimplemented!(),
+            Request::CommitBlock { block } => {
+                let (rsp_tx, rsp_rx) = oneshot::channel();
+
+                self.queue(QueuedBlock { block, rsp_tx });
```

Comment:
Duplicate blocks should be eliminated here:
https://github.com/ZcashFoundation/zebra/blob/126488e7a3ce2bdfacad6556d99178351fc30423/book/src/dev/rfcs/0005-state-updates.md#committing-non-finalized-blocks

Reply:
How do we want to handle this? Should we check if the incoming block is already present in the queue and if it is return a clone of the channel or should we just return an error? The former would require changing to a `broadcast` channel.

Reply:
We want to return a clone of the channel, because we want idempotent behaviour for block verification. (Returning an error and a success for duplicates of the same block could lead to weird bugs and race conditions.)

Reply:
This was already implicit in the design, which said:
> broadcast `Ok(block.hash())` via `block.rsp_tx`, and return

But it's a lot clearer after these fixes:
https://github.com/ZcashFoundation/zebra/pull/1136